java-lib/src/main/java/org/flinkextended/clink/feature/onehotencoder/ClinkOneHotEncoderModel.java

```diff
@@ -181,13 +187,24 @@ public void close() throws Exception {
     private static Pointer loadCppModel(
             Map<Param<?>, Object> paramMap, List<Tuple2<Integer, Integer>> modelDataList)
             throws IOException {
-        String paramString = ParamUtils.jsonEncode(paramMap);
+        File tmpDir = Files.createTempDirectory("ClinkOneHotEncoderModel").toFile();
+        String tmpDirStr = tmpDir.getAbsolutePath();
+
+        ParamUtils.saveMetadata(paramMap, ClinkOneHotEncoderModel.class, tmpDirStr);
```

Comment:
I agree that we should try to reduce duplication in our programs. `ClinkOneHotEncoderModel::save(...)` only constructs the stream graph to save model data, while `loadCppModel()` actually completes the job, so we cannot directly use `ClinkOneHotEncoderModel::save(...)` in `loadCppModel()`. Instead, I tried changing the implementation of `loadCppModel` to use `ByteArrayEncoder()` and `OneHotEncoderProtobufUtils.getModelDataByteArray()`, which have been used in `ClinkOneHotEncoderModel::save(...)`, and this might help to achieve the goal to reduce duplication.

During the fix of this problem I also found and fixed another bug in the saving process: while `loadCppModel()` and `OneHotEncoderModel::load` only saves or reads the model data, `ClinkOneHotEncoderModel::save(...)` saves both the length of model data and the model data itself. I unified them to `ClinkOneHotEncoderModel::save(...)`'s practice. You may notice the related change in `one_hot_encoder.cc`.

Reply:
Could we change `ClinkOneHotEncoderModel::save(...)` to use `ParamUtils.saveMetadata(...)` as well for better consistency?

Would it be better to rename `ParamUtils` to `ClinkReadWriteUtils` so that the relationship between these two utils is more self-explanatory?

Suppose user runs the following code, the current PR will save and read the entire model data again during `map(..)`, which is inefficient when the model data size is large. Could we avoid this overhead, e.g. load the cppModel during `ClinkOneHotEncoderModel::load(...)`?

```
modelA.save(path_1);
ClinkOneHotEncoderModel modelB = ClinkOneHotEncoderModel::load(path_1)
Table outputTable = modeB.map(...)
```

Reply:
I agree to the points to make `ClinkOneHotEncoderModel::save(...)` use `ParamUtils.saveMetadata(...)` and to rename `ParamUtils` to `ClinkReadWriteUtils`.

As for the third point, `ClinkOneHotEncoderModel` should be able to judge whether itself is loaded from certain file path, and if it is true, avoid re-saving the model data. I'll make the change in this direction.

Reply:
Has this issue been resolved? It seems that the PR will still load model twice if user runs the code snippet described above?

Reply:
I thought it has been resolved as the program has been able to avoid re-saving model data if it has been saved before. The two loading process might be inevitable: one is invoked in Java program as to support `Model.getModelData()` method, the other is invoked in C++ program to initialize Clink operator. Given that the target of this PR is to remove APIs that transmit model data through memory, we cannot load the data just once and share it among Java and C++ programs.

Reply:
As discussed offline, Given that loading model data into C++ operators is inevitable, I'll change the implementation so that `ClinkOneHotEncoderModel` will not load model data into Java memory space unless APIs that require this information, i.e. `getModelData()`, is invoked.